CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+- Respond to every comment. (to the best of your judgement! For example if the reviewer has simply complimented a smart line of code, it is ok to simply resolve it.)
+- Comment on individual lines of the PR yourself if you want to bring special attention to it for feedback.
+- If you are a core contributor and wish to push a large feature that would require multiple PRs, the recommended workflow is to first create a "feature" branch off of `main`. Then, make a PR that commits the architecture of this feature. This could simply be a function that has function stubs like 
+	`do_x()`
+	`do_y()` 
+	without implementing the functions themselves. Then, make PRs that would handle `do_x()` and `do_y()` seperately. Request a review for each PR, and when the entire feature is complete it is easier to merge that into the `main` branch.
+
+**When reviewing a PR:**
+
+- Don't accept code that doesn't pass all tests.
+- Don't accept commented out code.
+- Don't accept code that has not been unit tested.
+- Don't accept functions without documentation.
+- If you see a bug in the PR, don't merge and fix the bug with another PR. Comment on the PR for the person putting in the PR to fix.
+- If the person submitting the PR **is** a core contributor, do not merge the PR yourself, just approve it. The person submitting the PR should merge it.
+- If the person submitting the PR **is not** a core contributor, the reviewer should merge the PR. 
+- In the case that a PR is "good enough for now" but it is known that it will induce future changes, document that foreseen problem as an **Issue** immediately.
+- Be judicious with any comments in the code that you don't understand.
+
+**Guidelines for both the PR submitter and PR reviewer:**
+
+- Emoji usage is encouraged, especially if you think it conveys tone 🙂
+
+### Style Guide
+* State the types of the arguments in a function whenever possible eg. `function foo(x::Int)`. If you intend the type to be anything, make that explicit with `Any` eg. `function foo(x::Any)`.
+* State the return type of the function whenever possible eg. `function add_two(num::Int)::Int`.
+* Append ! to names of functions that modify their arguments.
+* Avoid using global variables. If you absolutely, absolutely must, use all caps eg. `const UNICORN = 4`.
+* Use `in` instead of `=` in for loop iterations, eg. use `for i in 1:5` instead of `for i = 1:5`.
+* Separate positional arguments from keyword arguments using a semicolon (;).
+* Explicitly state `return` in your functions eg. `return x` instead of just `x`, which would also work.
+
+This contributing document has been partially been derived from [its Python sibling](https://github.com/mggg/GerryChain/blob/master/CONTRIBUTING.md), and the Style Guide has been adopted from [the official Style Guide](https://help.github.com/articles/creating-a-pull-request/) and [jrevel's style guide.](https://docs.julialang.org/en/v1/manual/style-guide/index.html)
```

Comment:
Thank you for crediting folks! ✨ Here, I think you may have linked the GitHub help article instead of the official style guide. Similarly, the link that goes along with jrevel's style guide is a link to the Julia official style guide.

Reply:
sloppy! changing this rn